File: BodyMass-main/BodyMass/app/src/main/java/com/example/bodymass/MainActivity.java
package com.example.bodymass;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.RadioButton;
import android.widget.RadioGroup;
import android.widget.RelativeLayout;
import android.widget.TableLayout;
import android.widget.TextView;
import android.widget.Toast;

import com.google.android.material.textfield.TextInputLayout;

public class MainActivity extends AppCompatActivity {

    private RadioGroup rGroup;
    private RadioGroup rGroup2;
    private RadioGroup rGroup3;
    private RadioButton male;
    private RadioButton small;
    private TextInputLayout textInput;
    private TextInputLayout textInput2;
    private TableLayout table;
    private TextView textView2;
    private RelativeLayout relative;
    private Gender gender;
    private Morphology morphology;
    private int seleccion;



    @Override
    protected void onCreate(Bundle savedInstanceState) {




        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        rGroup = findViewById(R.id.rGroup);
        rGroup2 = findViewById(R.id.rGroup2);
        rGroup3 = findViewById(R.id.rGroup3);
        RadioButton relativeFat = findViewById(R.id.relativeFat);
        RadioButton wanDer = findViewById(R.id.wanDer);
        RadioButton loretz = findViewById(R.id.loretz);
        RadioButton creff = findViewById(R.id.creff);
        male = findViewById(R.id.male);
        RadioButton female = findViewById(R.id.female);
        small = findViewById(R.id.small);
        RadioButton medium = findViewById(R.id.medium);
        RadioButton broad = findViewById(R.id.broad);
        textInput = findViewById(R.id.textInput);
        textInput2 = findViewById(R.id.textInput2);
        table = findViewById(R.id.tabla);
        textView2 = findViewById(R.id.textView2);
        Button button = findViewById(R.id.button);
        relative = findViewById(R.id.relative);

         rGroup3.setOnCheckedChangeListener(new RadioGroup.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(RadioGroup group, int checkedId) {
                int radioSelected = rGroup3.getCheckedRadioButtonId();
                switch (radioSelected) {
                    case R.id.relativeFat:
                        relative.setAlpha(1);
                        textInput2.setVisibility(View.VISIBLE);
                        rGroup.setVisibility(View.VISIBLE);
                        rGroup2.setVisibility(View.GONE);
                        male.setChecked(true);
                        textInput2.setHint(R.string.waist_circumference_cm);
                        gender = Gender.MALE;
                        textView2.setAlpha(0);
                        seleccion = 1;
                        break;
                    case R.id.wanDer:
                        relative.setAlpha(1);
                        textInput2.setVisibility(View.GONE);
                        rGroup.setVisibility(View.VISIBLE);
                        rGroup2.setVisibility(View.GONE);
                        male.setChecked(true);
                        table.setAlpha(0);
                        gender = Gender.MALE;
                        textView2.setAlpha(0);
                        seleccion = 2;
                        break;
                    case R.id.loretz:
                        relative.setAlpha(1);
                        textInput2.setVisibility(View.VISIBLE);
                        rGroup.setVisibility(View.VISIBLE);
                        rGroup2.setVisibility(View.GONE);
                        textInput2.setHint(R.string.age);
                        table.setAlpha(0);
                        male.setChecked(true);
                        gender = Gender.MALE;
                        textView2.setAlpha(0);
                        seleccion = 3;
                        break;
                    case R.id.creff:
                        relative.setAlpha(1);
                        textInput2.setVisibility(View.VISIBLE);
                        rGroup.setVisibility(View.GONE);
                        rGroup2.setVisibility(View.VISIBLE);
                        textInput2.setHint(R.string.age);
                        table.setAlpha(0);
                        small.setChecked(true);
                        morphology = Morphology.SMALL;
                        textView2.setAlpha(0);
                        seleccion = 4;
                        break;
                }
            }
        });

        rGroup2.setOnCheckedChangeListener(new RadioGroup.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(RadioGroup group, int checkedId) {
                int radioSelected = rGroup2.getCheckedRadioButtonId();
                switch (radioSelected) {
                    case R.id.small:
                        morphology = Morphology.SMALL;
                        break;
                    case R.id.medium:
                        morphology = Morphology.MEDIUM;
                        break;
                    case R.id.broad:
                        morphology = Morphology.BROAD;
                        break;
                }
            }
        });

        rGroup.setOnCheckedChangeListener(new RadioGroup.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(RadioGroup group, int checkedId) {
                int radioSelected = rGroup.getCheckedRadioButtonId();
                switch (radioSelected) {
                    case R.id.male:
                        gender = Gender.MALE;
                        break;
                    case R.id.female:
                        gender = Gender.FEMALE;
                        break;
                }
            }
        });

        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                double resultado;
                double constante;
                if (textInput.getEditText().getText().toString().isEmpty()) {
                    Toast.makeText(getApplicationContext(), "Error en la altura", Toast.LENGTH_SHORT).show();
                } else {
                    switch (seleccion) {
                        case 1:
                            if (textInput2.getEditText().getText().toString().isEmpty()) {
                                Toast.makeText(getApplicationContext(), "Error en la cintura", Toast.LENGTH_SHORT).show();
                            }
                            else{
                                if (gender == Gender.MALE) {
                                    constante = Constants.N_hombre;
                                } else {
                                    constante = Constants.N_mujer;
                                }
                                resultado = constante - 20 * Double.parseDouble(textInput.getEditText().getText().toString()) / Double.parseDouble(textInput2.getEditText().getText().toString());
                                resultado = Math.round(resultado*100.0)/100.0;
                                textView2.setAlpha(1);
                                textView2.setText(resultado + "%");
                                table.setAlpha(1);
                            }
                            break;
                        case 2:
                            if (gender == Gender.MALE) {
                                constante = Constants.M_hombre;
                            } else {
                                constante = Constants.M_muujer;
                            }
                            resultado = (Double.parseDouble(textInput.getEditText().getText().toString()) - 150) * constante + 50;
                            resultado = Math.round(resultado*100.0)/100.0;
                            textView2.setAlpha(1);
                            textView2.setText(resultado + "Kg");
                            break;
                        case 3:
                            if (textInput2.getEditText().getText().toString().isEmpty()) {
                                Toast.makeText(getApplicationContext(), "Error en la edad", Toast.LENGTH_SHORT).show();
                            }else{
                                if (gender == Gender.MALE) {
                                    constante = Constants.K_hombre;
                                } else {
                                    constante = Constants.K_muujer;
                                }
                                resultado = Double.parseDouble(textInput.getEditText().getText().toString()) - 100 - (Double.parseDouble(textInput.getEditText().getText().toString()) - 150) / 4 + (Double.parseDouble(textInput2.getEditText().getText().toString())-20)/constante;
                                resultado = Math.round(resultado*100.0)/100.0;
                                textView2.setAlpha(1);
                                textView2.setText(resultado + "Kg");
                            }
                            break;
                        case 4:
                            if (textInput2.getEditText().getText().toString().isEmpty()) {
                                Toast.makeText(getApplicationContext(), "Error en la edad", Toast.LENGTH_SHORT).show();
                            }
                            else{
                                resultado = (Double.parseDouble(textInput.getEditText().getText().toString())-100+Double.parseDouble(textInput2.getEditText().getText().toString())/10);
                                if (morphology == Morphology.BROAD) {
                                    resultado = resultado * 0.9 * 1.1;
                                } else if (morphology == Morphology.SMALL) {
                                    resultado = resultado * 0.9 * 0.9;
                                } else {
                                    resultado = resultado * 0.9;
                                }
                                resultado = Math.round(resultado*100.0)/100.0;
                                textView2.setAlpha(1);
                                textView2.setText(resultado + "Kg");
                            }
                            break;
                    }
                }
            }
        });








    }
}

File: BodyMassKotlin/app/src/main/java/com/example/bodymasskotlin/MainActivity.kt
package com.example.bodymasskotlin

import android.annotation.SuppressLint
import android.os.Bundle
import android.view.View
import android.widget.*
import androidx.appcompat.app.AppCompatActivity
import com.google.android.material.textfield.TextInputLayout
import kotlin.math.roundToInt


class MainActivity : AppCompatActivity() {

    private lateinit var rGroup: RadioGroup
    private lateinit var rGroup2: RadioGroup
    private lateinit var rGroup3: RadioGroup
    private lateinit var male: RadioButton
    private lateinit var small: RadioButton
    private lateinit var textInput: TextInputLayout
    private lateinit var textInput2: TextInputLayout
    private lateinit var table: TableLayout
    private lateinit var textView2: TextView
    private lateinit var relative: RelativeLayout
    private lateinit var gender: Gender
    private lateinit var morphology: Morphology
    private var seleccion: Int = 0


    @SuppressLint("SetTextI18n")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        rGroup = findViewById(R.id.rGroup)
        rGroup2 = findViewById(R.id.rGroup2)
        rGroup3 = findViewById(R.id.rGroup3)
        findViewById<RadioButton>(R.id.relativeFat)
        findViewById<RadioButton>(R.id.wanDer)
        findViewById<RadioButton>(R.id.loretz)
        findViewById<RadioButton>(R.id.creff)
        male = findViewById(R.id.male)
        findViewById<RadioButton>(R.id.female)
        small = findViewById(R.id.small)
        findViewById<RadioButton>(R.id.medium)
        findViewById<RadioButton>(R.id.broad)
        textInput = findViewById(R.id.textInput)
        textInput2 = findViewById(R.id.textInput2)
        table = findViewById(R.id.tabla)
        textView2 = findViewById(R.id.textView2)
        val button: Button = findViewById(R.id.button)
        relative = findViewById(R.id.relative)
        rGroup3.setOnCheckedChangeListener { _, _ ->
            when (rGroup3.checkedRadioButtonId) {
                R.id.relativeFat -> {
                    relative.alpha = 1f
                    textInput2.visibility = View.VISIBLE
                    rGroup.visibility = View.VISIBLE
                    rGroup2.visibility = View.GONE
                    male.isChecked = true
                    textInput2.setHint(R.string.waist_circumference_cm)
                    gender = Gender.MALE
                    textView2.alpha = 0f
                    seleccion = 1
                }
                R.id.wanDer -> {
                    relative.alpha = 1f
                    textInput2.visibility = View.GONE
                    rGroup.visibility = View.VISIBLE
                    rGroup2.visibility = View.GONE
                    male.isChecked = true
                    table.alpha = 0f
                    gender = Gender.MALE
                    textView2.alpha = 0f
                    seleccion = 2
                }
                R.id.loretz -> {
                    relative.alpha = 1f
                    textInput2.visibility = View.VISIBLE
                    rGroup.visibility = View.VISIBLE
                    rGroup2.visibility = View.GONE
                    textInput2.setHint(R.string.age)
                    table.alpha = 0f
                    male.isChecked = true
                    gender = Gender.MALE
                    textView2.alpha = 0f
                    seleccion = 3
                }
                R.id.creff -> {
                    relative.alpha = 1f
                    textInput2.visibility = View.VISIBLE
                    rGroup.visibility = View.GONE
                    rGroup2.visibility = View.VISIBLE
                    textInput2.setHint(R.string.age)
                    table.alpha = 0f
                    small.isChecked = true
                    morphology = Morphology.SMALL
                    textView2.alpha = 0f
                    seleccion = 4
                }
            }
        }
        rGroup2.setOnCheckedChangeListener { _, _ ->
            when (rGroup2.checkedRadioButtonId) {
                R.id.small -> morphology = Morphology.SMALL
                R.id.medium -> morphology = Morphology.MEDIUM
                R.id.broad -> morphology = Morphology.BROAD
            }
        }
        rGroup.setOnCheckedChangeListener { _, _ ->
            when (rGroup.checkedRadioButtonId) {
                R.id.male -> gender = Gender.MALE
                R.id.female -> gender = Gender.FEMALE
            }
        }
        button.setOnClickListener {
            var resultado: Double
            val constante: Double
            if (textInput.editText!!.text.toString().isEmpty()) {
                Toast.makeText(applicationContext, "Error en la altura", Toast.LENGTH_SHORT)
                    .show()
            } else {
                when (seleccion) {
                    1 -> if (textInput2.editText!!.text.toString().isEmpty()) {
                        Toast.makeText(
                            applicationContext,
                            "Error en la cintura",
                            Toast.LENGTH_SHORT
                        ).show()
                    } else {
                        constante = if (gender === Gender.MALE) {
                            Constants.N_hombre
                        } else {
                            Constants.N_mujer
                        }
                        resultado = constante - 20 * textInput.editText!!.text.toString()
                            .toDouble() / textInput2.editText!!
                            .text.toString().toDouble()
                        resultado = (resultado * 100.0).roundToInt() / 100.0
                        textView2.alpha = 1f
                        textView2.text = "$resultado%"
                        table.alpha = 1f
                    }
                    2 -> {
                        constante = if (gender === Gender.MALE) {
                            Constants.M_hombre
                        } else {
                            Constants.M_muujer
                        }
                        resultado = (textInput.editText!!.text.toString()
                            .toDouble() - 150) * constante + 50
                        resultado = (resultado * 100.0).roundToInt() / 100.0
                        textView2.alpha = 1f
                        textView2.text = resultado.toString() + "Kg"
                    }
                    3 -> if (textInput2.editText!!.text.toString().isEmpty()) {
                        Toast.makeText(
                            applicationContext,
                            "Error en la edad",
                            Toast.LENGTH_SHORT
                        ).show()
                    } else {
                        constante = if (gender === Gender.MALE) {
                            Constants.K_hombre
                        } else {
                            Constants.K_muujer
                        }
                        resultado = textInput.editText!!.text.toString()
                            .toDouble() - 100 - (textInput.editText!!
                            .text.toString().toDouble() - 150) / 4 + (textInput2.editText!!
                            .text.toString().toDouble() - 20) / constante
                        resultado = (resultado * 100.0).roundToInt() / 100.0
                        textView2.alpha = 1f
                        textView2.text = resultado.toString() + "Kg"
                    }
                    4 -> if (textInput2.editText!!.text.toString().isEmpty()) {
                        Toast.makeText(
                            applicationContext,
                            "Error en la edad",
                            Toast.LENGTH_SHORT
                        ).show()
                    } else {
                        resultado = textInput.editText!!.text.toString()
                            .toDouble() - 100 + textInput2.editText!!
                            .text.toString().toDouble() / 10
                        resultado = when {
                            morphology === Morphology.BROAD -> {
                                resultado * 0.9 * 1.1
                            }
                            morphology === Morphology.SMALL -> {
                                resultado * 0.9 * 0.9
                            }
                            else -> {
                                resultado * 0.9
                            }
                        }
                        resultado = (resultado * 100.0).roundToInt() / 100.0
                        textView2.alpha = 1f
                        textView2.text = resultado.toString() + "Kg"
                    }
                }
            }
        }
    }
}
